hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1323,29 +1323,36 @@ public OmMultipartUploadList listMultipartUploads(String volumeName,
 
       List<OmMultipartUpload> collect = multipartUploadKeys.stream()
           .map(OmMultipartUpload::from)
-          .map(upload -> {
+          .peek(upload -> {
             String dbKey = metadataManager
                 .getOzoneKey(upload.getVolumeName(),
                     upload.getBucketName(),
                     upload.getKeyName());
             try {
-              Table<String, OmKeyInfo> openKeyTable =
-                  metadataManager.getOpenKeyTable();
+              Table<String, OmMultipartKeyInfo> keyInfoTable =
+                  metadataManager.getMultipartInfoTable();
 
-              OmKeyInfo omKeyInfo =
-                  openKeyTable.get(upload.getDbKey());
+              OmMultipartKeyInfo multipartKeyInfo =
+                  keyInfoTable.get(upload.getDbKey());
 
               upload.setCreationTime(
-                  Instant.ofEpochMilli(omKeyInfo.getCreationTime()));
-
-              upload.setReplicationType(omKeyInfo.getType());
-              upload.setReplicationFactor(omKeyInfo.getFactor());
+                  Instant.ofEpochMilli(multipartKeyInfo.getCreationTime()));
+
+              TreeMap<Integer, PartKeyInfo> partKeyInfoMap =
+                  multipartKeyInfo.getPartKeyInfoMap();
+              if (!partKeyInfoMap.isEmpty()) {
+                PartKeyInfo partKeyInfo =
```

Comment:
If there are no parts, if partKeyInfoMap.isEmpty()?

Reply:
Hi @bharatviswa504 not sure I understand, the map is not empty we get the type and factor from the first part. isEmpty() => size == 0

Reply:
If the MPU uploaded does not have any parts then I don't see that logic here.